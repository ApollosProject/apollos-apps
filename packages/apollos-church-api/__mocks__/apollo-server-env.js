import fetch from 'jest-fetch-mock';

import { Constants } from 'apollos-church-api/src/connectors/rock';
import * as apolloDatasourceMocks from './apollo-datasource-mocks';
import * as rockMocks from './rock-api-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('/api/v1/events/current')) {
      return resolveWith(apolloDatasourceMocks.liveStreamLive(), request.url);
    }
    if (request.url.match('/passage/html')) {
      return resolveWith(apolloDatasourceMocks.ESVScripture(), request.url);
    }
    return Promise.reject();
  });
};

fetch.mockRockDataSourceAPI = () => {
  fetch.mockImplementation((request) => {
    let { url } = request;
    url = decodeURI(url);
    if (!url.match(Constants.ROCK_API)) return Promise.reject();

    if (url.match('api/ContentChannels/\\d')) {
      return resolveWith(rockMocks.contentChannel());
    }

    if (url.match('api/ContentChannels')) {
      return resolveWith([rockMocks.contentChannel()]);
    }

    if (url.match('api/ContentChannelItems/test-case-no-attributes')) {
      const mock = rockMocks.contentItem();
      mock.AttributeValues = {};
      mock.Attributes = {};
      return resolveWith(mock);
    }

    if (url.match('api/ContentChannelItems/\\d')) {
      return resolveWith(rockMocks.contentItem());
    }

    if (url.match('api/ContentChannelItems')) {
      return resolveWith([rockMocks.contentItem()]);
    }

    if (url.match('api/ContentChannelItemAssociations')) {
      return resolveWith([rockMocks.contentChannelItemAssociation()]);
    }

    if (url.match('api/Auth/Login')) {
      const body = JSON.parse(request.body);
      const response = new Response('');
      if (body.Password === 'good') {
        response.headers.set('set-cookie', 'some cookie');
        return Promise.resolve(response);
      }
      return Promise.reject({ ...response, status: 401 });
    }

    if (url.match('api/People/GetCurrentPerson')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People/\\d')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People')) {
      if (request.method === 'POST') {
        const { Email } = JSON.parse(request.body);
        if (!Email) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ personId: 35 });
      }

      return resolveWith([rockMocks.people()]);
    }

    if (url.match('api/UserLogins')) {
      if (request.method === 'POST') {
        const { UserName } = JSON.parse(request.body);
        if (!UserName) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ id: 21 });
      }
      const identity = url // identity = UserName
        .split('eq')
        .pop()
        .trim(' '); // EXAMPLE URL: /api/UserLogins?$filter=UserName eq 'isaac.hardy@newspring.cc'

      if (!identity) {
        const response = new Response('');
        response.status = 400;
        return Promise.reject(response);
      }
      if (identity === `'isaac.hardy@newspring.cc'`)
        return resolveWith([rockMocks.userLogins()]);

      return resolveWith([]);
    }

    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
