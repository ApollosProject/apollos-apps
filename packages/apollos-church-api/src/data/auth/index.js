import { gql } from 'apollo-server';
import { createGlobalId } from '../node';

// export { default as model } from './model';
export { default as dataSource } from './data-source';

export const schema = gql`
  type AuthenticatedUser {
    id: ID!
    profile: Person
  }

  type Authentication {
    user: AuthenticatedUser
    token: String
  }

  extend type Mutation {
    authenticate(identity: String!, password: String!): Authentication
    registerPerson(email: String!, password: String!): Authentication
  }

  extend type Query {
    currentUser: AuthenticatedUser
  }
`;

export const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};
