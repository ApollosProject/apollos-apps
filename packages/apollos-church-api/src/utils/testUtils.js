import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
