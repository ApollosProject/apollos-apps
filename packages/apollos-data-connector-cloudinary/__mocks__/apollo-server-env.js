import fetch from 'jest-fetch-mock';

import * as apolloDatasourceMocks from './apollo-datasource-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('api.scripture.api.bible')) {
      return resolveWith(apolloDatasourceMocks.Scripture(), request.url);
    }
    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
