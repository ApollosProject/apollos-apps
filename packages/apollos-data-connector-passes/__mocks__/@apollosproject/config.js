const ApolloServer = require.requireActual('@apollosproject/config').default;

// These example certs and private key were taken from easily found public examples on the web.
// I know not what they actaully represent!
const examplePem =
  '\n-----BEGIN CERTIFICATE-----\nMIICMzCCAZygAwIBAgIJALiPnVsvq8dsMA0GCSqGSIb3DQEBBQUAMFMxCzAJBgNV\nBAYTAlVTMQwwCgYDVQQIEwNmb28xDDAKBgNVBAcTA2ZvbzEMMAoGA1UEChMDZm9v\nMQwwCgYDVQQLEwNmb28xDDAKBgNVBAMTA2ZvbzAeFw0xMzAzMTkxNTQwMTlaFw0x\nODAzMTgxNTQwMTlaMFMxCzAJBgNVBAYTAlVTMQwwCgYDVQQIEwNmb28xDDAKBgNV\nBAcTA2ZvbzEMMAoGA1UEChMDZm9vMQwwCgYDVQQLEwNmb28xDDAKBgNVBAMTA2Zv\nbzCBnzANBgkqhkiG9w0BAQEFAAOBjQAwgYkCgYEAzdGfxi9CNbMf1UUcvDQh7MYB\nOveIHyc0E0KIbhjK5FkCBU4CiZrbfHagaW7ZEcN0tt3EvpbOMxxc/ZQU2WN/s/wP\nxph0pSfsfFsTKM4RhTWD2v4fgk+xZiKd1p0+L4hTtpwnEw0uXRVd0ki6muwV5y/P\n+5FHUeldq+pgTcgzuK8CAwEAAaMPMA0wCwYDVR0PBAQDAgLkMA0GCSqGSIb3DQEB\nBQUAA4GBAJiDAAtY0mQQeuxWdzLRzXmjvdSuL9GoyT3BF/jSnpxz5/58dba8pWen\nv3pj4P3w5DoOso0rzkZy2jEsEitlVM2mLSbQpMM+MUVQCQoiG6W9xuCFuxSrwPIS\npAqEAuV4DNoxQKKWmhVv+J0ptMWD25Pnpxeq5sXzghfJnslJlQND\n-----END CERTIFICATE-----';
const examplePrivateKey =
  '-----BEGIN RSA PRIVATE KEY-----\nProc-Type: 4,ENCRYPTED\nDEK-Info: DES-EDE3-CBC,E3B1C06E0D0C2633\n\ngvmXzl6W7eV1a3N5rQNwBWKY9on3IgxZudS33cip5f88FotsPSDJMvqj6LVw2RxobDjhlOOzqmTb\nVrlTnoQ6CogXFZSfiPmixiyyptCUEKJkSiEhYGM5GQm0OoGcLeLbgBb9tRpWh5IlXulKD6XFhx8q\n/eGg5a+mSkX1i7kv2+Ih3jHmEKwrnfzhcA29pBF3OQJo+Ks9IYneuk676pHtsIs7CpFKq1tDvD8Q\nO7URxnVnHLltaFvIxshqyZu92xbUYZR7YzjXl5+3w4TVgeAHUogEV+H9iZTosD/copUsbQO+78w2\nE1D3iDS94wRgx0Tjv4xlwrTpOV38FS5rdL32492DcCRlCYM4VtuwjYeWi5shJg69jCb0EwGRqfAo\nxko+lbKWELTuFKwD7n1rc/2fTarbGuf8S2AEggBLZyfXHC/9N84mXLFO2XKq+0WdiEFhQj2Cze+a\n9qcSK6tPSrjK1LPlnOOppFgDElZaZ0rxsgjtiWSIAEw/Ad+SIM5u+vqwzF8J317JlsdKoBFDw8mS\nMxCMuMksKJ23mgvY+THRIVgH3E7lEDZQzCi1Uy6ldLJcran/6wHwP88pVM2odiHkpnrJGcEBbbIk\nqsxJZhFT8aUt/cUEBj3fnP7cxoNLQfTHMPqUTqKBWaVufFzGU9YB1R+XWFULLddwJHnV7gPheBlk\nMDapowb+Is77+a9Y2VDsOXEvNpqTY0giiSrckG05IZnrhJ24JnSCwyNd99lm7XKdEGGrjBCMqIyI\nFqox8Ahkv3KWAJPYK1eOCc5d/KwZHlnlFJq7ZYy9u3fEnxQCjOEmeXLkLangKA==\n-----END RSA PRIVATE KEY-----';

const privateKeyPassword = 'password';

ApolloServer.loadJs({
  APP: {
    ROOT_API_URL: 'http://localhost:4000',
  },
  PASS: {
    TEMPLATES: {
      EXAMPLE: `${process.env.PWD}/example-pass.pass`,
    },
    CERTIFICATES: {
      WWDR: examplePem,
      SIGNER_CERT: examplePem,
      SIGNER_KEY: examplePrivateKey,
      SIGNER_KEY_PASSPHRASE: privateKeyPassword,
    },
  },
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
  },
});

export default ApolloServer;
