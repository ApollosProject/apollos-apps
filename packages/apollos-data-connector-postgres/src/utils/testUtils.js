/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
