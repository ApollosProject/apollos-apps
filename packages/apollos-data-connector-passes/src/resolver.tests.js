import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import { authSchema } from '@apollosproject/data-schema';
import { Person } from '@apollosproject/data-connector-rock';
import { createGlobalId } from '@apollosproject/server-core';
import gql from 'graphql-tag';

import { generateToken } from '@apollosproject/data-connector-rock/lib/auth';
import * as Pass from './index';

class AuthDataSourceMock {
  initialize = () => {};

  getCurrentPerson = () => ({
    id: 51,
    firstName: 'Isaac',
    lastName: 'Hardy',
    nickName: 'Isaac',
    email: 'isaac.hardy@newspring.cc',
    photo: {
      url:
        'https://apollosrock.newspring.cc:443/GetImage.ashx?guid=60fd5f35-3167-4c26-9a30-d44937287b87',
    },
  });

  getCurrentPersonAlternateLookupId = () => '0faad2f-3258f47';
}

const Auth = {
  schema: authSchema,
  dataSource: AuthDataSourceMock,
  resolver: { Query: { currentUser: () => ({ profile: { id: 51 } }) } },
};

const { getContext, getSchema } = createTestHelpers({
  Pass,
  Person,
  Auth,
  Theme: {
    schema: gql`
      scalar Color
    `,
  },
});

describe('Passes', () => {
  let schema;
  let context;
  beforeEach(() => {
    schema = getSchema();
    context = getContext();

    fetch.resetMocks();
    fetch.mockLiveDataSourceApis();
  });

  describe('as a logged in user', () => {
    beforeEach(() => {
      const token = generateToken({ cookie: 'some-cookie', sessionId: 123 });
      context = getContext({
        req: {
          headers: { authorization: token },
        },
      });
    });

    it('queries by node', async () => {
      const id = createGlobalId('EXAMPLE', 'Pass');
      const query = `
        query {
          node(id: "${id}") {
            ...on Pass {
              id
              type
              description
              logo { uri }
              thumbnail { uri }
              barcode { uri }
              primaryFields {
                key
                label
                value
                textAlignment
              }
              secondaryFields {
                key
                label
                value
                textAlignment
              }
              backgroundColor
              foregroundColor
              labelColor
              logoText
              passkitFileUrl
            }
          }
        }
      `;
      const rootValue = {};

      console.log({ context });

      const result = await graphql(schema, query, rootValue, context);
      expect(result).toMatchSnapshot();
    });

    it('returns the checkin pass', async () => {
      const query = `
        query {
          userPass {
            id
            type
            description
            logo { uri }
            thumbnail { uri }
            barcode { uri }
            primaryFields {
              key
              label
              value
              textAlignment
            }
            secondaryFields {
              key
              label
              value
              textAlignment
            }
            backgroundColor
            foregroundColor
            labelColor
            logoText
            passkitFileUrl
          }
        }
      `;
      const rootValue = {};
      const result = await graphql(schema, query, rootValue, context);
      expect(result).toMatchSnapshot();
    });
  });
});
