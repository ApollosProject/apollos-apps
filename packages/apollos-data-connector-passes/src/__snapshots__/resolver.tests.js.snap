// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes as a logged in user queries by node 1`] = `
Object {
  "data": Object {
    "node": Object {
      "backgorundColor": null,
      "barcode": Object {
        "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAATvSURBVO3BQY4bSRAEwfAC//9lXx3zVECjk7MaIczwj1QtOaladFK16KRq0UnVopOqRSdVi06qFp1ULTqpWnRSteikatFJ1aKTqkUnVYtOqhZ98hKQn6RmAnKj5gbIpOYGyDepmYD8JDVvnFQtOqladFK16JNlajYB+SY1N0Bu1ExAbtS8oWYTkE0nVYtOqhadVC365MuAPKHmCTUTkBsgk5oJyBtqboBMat4A8oSabzqpWnRSteikatEnvxyQGyCTmhs1bwCZ1Exq/mUnVYtOqhadVC365JdTMwG5AfKEmjeATGr+ZSdVi06qFp1ULfrky9T8JDWbgExqbtRMQCYgk5o31PxNTqoWnVQtOqla9MkyID8JyKRmAjKpmYBMap4AMqm5UTMBmdTcAPmbnVQtOqladFK16JOX1PwmQG6AvAFkUjMBmdTcqPlNTqoWnVQtOqla9MlLQCY1E5AbNROQJ9Q8oWYC8gSQSc0EZAIyqXkCyKTmBsikZgJyo+aNk6pFJ1WLTqoW4R/5hwF5Qs0TQCY1E5A31ExAbtQ8AWRS88ZJ1aKTqkUnVYs+eQnIG2pugExqboBMaiYgk5obIJOaGyA3am6APKHmBsiNmk0nVYtOqhadVC3CP7IIyBtq3gDyhJoJyI2aJ4BMaiYgN2q+Ccik5o2TqkUnVYtOqhZ98mVqngDyhJobNTdAJjVPALlRMwGZ1HwTkBs1m06qFp1ULTqpWvTJMjU3QJ5QcwPkCSCTmgnIpGYC8gSQN4DcqJmA3KiZgExq3jipWnRSteikatEnLwGZ1ExAJjU3QCYgN2o2qZmATGomIDdqJiATkCfU3Kj5P51ULTqpWnRSteiTZUAmNROQJ9TcAJnUvAFkUjMBmdTcALlR8wSQSc0EZFLzk06qFp1ULTqpWvTJS2omIBOQGzUTkAnIjZoJyBtqbtS8oeYGyKRmUjMB+ZucVC06qVp0UrXok2VqboBMQCY1N0Bu1NwAuQEyqZmAPKHmBsgTQCY1E5AbIJOaTSdVi06qFp1ULfrkJSCTmgnIjZobIE8AuVFzo+YJNROQJ9Q8oWYCMqmZgExqvumkatFJ1aKTqkWfvKTmCTU3QCY1E5An1ExAJjVPqJmAPAHkRs0NkEnNBOQGyI2aN06qFp1ULTqpWoR/ZBGQN9RMQN5QcwPkRs0NkL+JmgnIpOabTqoWnVQtOqla9MkyNZvU3ACZ1ExAJjVPAJnU3KiZgDyh5gkgN2pugExq3jipWnRSteikatEnLwH5SWqeUHOjZgIyqXkCyKRmAvIEkEnNDZBJzQRkUrPppGrRSdWik6pFnyxTswnIG0Bu1Exq3lAzAZnUTEBu1Dyh5v90UrXopGrRSdWiT74MyBNqnlAzAblR8wSQSc2kZgIyqXkCyCYgP+mkatFJ1aKTqkWf/HJAngAyqXkDyA2QN9RMQG6ATGpugExq3jipWnRSteikatEn/xg1TwCZ1ExqbtTcAJnU3ACZgExqngAyqfmmk6pFJ1WLTqoWffJlar5JzQRkUjMBmdRMQN5Q8wSQGzVPAJnUTEAmNZtOqhadVC06qVr0yTIgPwnIpGYCcgNkUvNNQG7UPAFkUvN/OqladFK16KRqEf6RqiUnVYtOqhadVC06qVp0UrXopGrRSdWik6pFJ1WLTqoWnVQtOqladFK16KRq0UnVov8ACbpJRE/izd0AAAAASUVORK5CYII=",
      },
      "description": "Apollos Church",
      "foregroundColor": "rgb(248,247,244)",
      "id": "Pass:ee0dc76d80d041692cdd83b50820a5eb",
      "labelColor": null,
      "logo": Object {
        "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADoAAAA6CAYAAADhu0ooAAAAAXNSR0IArs4c6QAABGFJREFUaAXtWktPU0EUPuX9aHlaeYmIvCkUqEBcGA0GiVFCjAYE6sKdK9f8A3+BiTt3riioiIpBNBrDyoCUt4AaQ4yJUQmPUsrLGUxrezsz9869cyGVns29M2fOzPnmnDn3zGkNcOv2LhwCijgEGPcghoH+b5YOWzRs0RDdgbDrhqjhqGofGotGUbdAIONadSU8uNlJnXFodg4u37tP5Ytg7ItFb9TZmLo2FBdCVpKJOUYrU3egZqMRLpQVM/WMMBigo7aaOUYrU3egbTYrREXIL9NZW6MVC1NeXgOmuDzTLuO23hkqsrPAmpPpbQp/6gq05KgZbLk5ipWWO8uKJyIM1BUor+JttmqIROdVD9INKFa3nTPAZKLI21hSpAdO0A3oucICyE1N4Va6s06foKQb0M56usJ3BoZgdWODuAktVgsYY2OIPC2dugCNj46Gq1UVRL22dnbg7tth6B+fIvJZskQBhZ26AG2xliOrxBJVeDU7D79cLugeHSfycae9np1JUQUZDF2Asr6djtGxPXUGZz7CsttNVO1swUl0vpOJPLWdwoFmmOiRc3N7G/rGp/d09eB3J9l9DegT035KbEooHChWEOeuJHo5MwdL6+s+luOD0/cufbELTgmFA7UzPg/do4HAhjBw1z/g/mBLMzO4sip/WdK7UKCWrEyUr2aR1gHP1hb0T/x1W++ATRSBH49PeptBT97MKmgCvw6hQFmKvZgmBx/HSKCV/XSD1hplNx9/Gdq7MKD4XF63VdHWAdp5fD23AD/X1ohyZpMRmmTuskRBQqcwoOdRlSA7JYmwBIB7cxOeStzWOxAnEI+cdPcVFZSEAWVdnAemZlHK5/FiC3r2jNCTh+bKckiOiwuS4e0QAjQxJgauVFmoa0ujrXTgm/kF+LGyKu3ea8dGRQEurmklIUAxyAQElkQujweeT86QWL6+7d1deOic8LWlLyJuNEKAstz2GQLpQmdUjhyM3PdMQT6cSEuVm4LJ1ww0O9kEOBDRyCFJEmjj3i18hu/LKzS25iqhZqA45cO5KYnWUADCgUgJ7SD37R2jByXWRUHJ/JqBspKE/okpcKOMSCn1MNy30HwE6vNylU4VNE4T0CqU7pWjtI9GctFWKjf86Qt8W1qWdvvaWqyqCSjLmivorjmIknYewn+P6R1jp4TRCorhpDVVA8VlSVyepFGfcxo2ONzWOw+r8pCWmAAXLaXeoVxP1b+mNZYWQUaSkbrY4tKSqtIlDmvYG0yUbAh70RNKvYmqDGKoBmqvtbHmha6mBuiCBuYYNcxLyKKpCfHwm3KPpc2pynVNqPCFC2AHQdGRkdBabeVeWhVQXMqMQyXNgyI1KaEqoKzi9H6AP52fBwXp6VxLcQPFZUj8c8NBE69VuYF2CK7Oqd0wXqDcUbfZUkYtPKtVWo1cOvqm1h4/Bu+/LioSN4T/r6ton0JnEPcZDR1ogZqGgQbuR+i3whYNfRsGIghbNHA/Qr91aCz6BzYbARjcFUOvAAAAAElFTkSuQmCC",
      },
      "logoText": "Apollos Church",
      "passkitFileUrl": null,
      "primaryFields": Array [
        Object {
          "key": "member",
          "label": null,
          "textAlignment": "LEFT",
          "value": "{{ person.FullName }}",
        },
      ],
      "secondaryFields": Array [
        Object {
          "key": "home",
          "label": "home",
          "textAlignment": "LEFT",
          "value": "{{ person | Address:'Home','[[City]], [[State]]' }}",
        },
        Object {
          "key": "campus",
          "label": "campus",
          "textAlignment": "RIGHT",
          "value": "{% assign personCampus = person | Campus %}{{ personCampus.Name  }}",
        },
      ],
      "thumbnail": Object {
        "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=dd2aa592-80ac-4d7a-8bdc-47aeb25ce158&format=png&height=200",
      },
      "type": "GENERIC",
    },
  },
}
`;

exports[`Passes as a logged in user returns the checkin pass 1`] = `
Object {
  "data": Object {
    "userPasses": Array [
      Object {
        "backgorundColor": null,
        "barcode": Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAJQAAACUCAYAAAB1PADUAAAAAklEQVR4AewaftIAAATvSURBVO3BQY4bSRAEwfAC//9lXx3zVECjk7MaIczwj1QtOaladFK16KRq0UnVopOqRSdVi06qFp1ULTqpWnRSteikatFJ1aKTqkUnVYtOqhZ98hKQn6RmAnKj5gbIpOYGyDepmYD8JDVvnFQtOqladFK16JNlajYB+SY1N0Bu1ExAbtS8oWYTkE0nVYtOqhadVC365MuAPKHmCTUTkBsgk5oJyBtqboBMat4A8oSabzqpWnRSteikatEnvxyQGyCTmhs1bwCZ1Exq/mUnVYtOqhadVC365JdTMwG5AfKEmjeATGr+ZSdVi06qFp1ULfrky9T8JDWbgExqbtRMQCYgk5o31PxNTqoWnVQtOqla9MkyID8JyKRmAjKpmYBMap4AMqm5UTMBmdTcAPmbnVQtOqladFK16JOX1PwmQG6AvAFkUjMBmdTcqPlNTqoWnVQtOqla9MlLQCY1E5AbNROQJ9Q8oWYC8gSQSc0EZAIyqXkCyKTmBsikZgJyo+aNk6pFJ1WLTqoW4R/5hwF5Qs0TQCY1E5A31ExAbtQ8AWRS88ZJ1aKTqkUnVYs+eQnIG2pugExqboBMaiYgk5obIJOaGyA3am6APKHmBsiNmk0nVYtOqhadVC3CP7IIyBtq3gDyhJoJyI2aJ4BMaiYgN2q+Ccik5o2TqkUnVYtOqhZ98mVqngDyhJobNTdAJjVPALlRMwGZ1HwTkBs1m06qFp1ULTqpWvTJMjU3QJ5QcwPkCSCTmgnIpGYC8gSQN4DcqJmA3KiZgExq3jipWnRSteikatEnLwGZ1ExAJjU3QCYgN2o2qZmATGomIDdqJiATkCfU3Kj5P51ULTqpWnRSteiTZUAmNROQJ9TcAJnUvAFkUjMBmdTcALlR8wSQSc0EZFLzk06qFp1ULTqpWvTJS2omIBOQGzUTkAnIjZoJyBtqbtS8oeYGyKRmUjMB+ZucVC06qVp0UrXok2VqboBMQCY1N0Bu1NwAuQEyqZmAPKHmBsgTQCY1E5AbIJOaTSdVi06qFp1ULfrkJSCTmgnIjZobIE8AuVFzo+YJNROQJ9Q8oWYCMqmZgExqvumkatFJ1aKTqkWfvKTmCTU3QCY1E5An1ExAJjVPqJmAPAHkRs0NkEnNBOQGyI2aN06qFp1ULTqpWoR/ZBGQN9RMQN5QcwPkRs0NkL+JmgnIpOabTqoWnVQtOqla9MkyNZvU3ACZ1ExAJjVPAJnU3KiZgDyh5gkgN2pugExq3jipWnRSteikatEnLwH5SWqeUHOjZgIyqXkCyKRmAvIEkEnNDZBJzQRkUrPppGrRSdWik6pFnyxTswnIG0Bu1Exq3lAzAZnUTEBu1Dyh5v90UrXopGrRSdWiT74MyBNqnlAzAblR8wSQSc2kZgIyqXkCyCYgP+mkatFJ1aKTqkWf/HJAngAyqXkDyA2QN9RMQG6ATGpugExq3jipWnRSteikatEn/xg1TwCZ1ExqbtTcAJnU3ACZgExqngAyqfmmk6pFJ1WLTqoWffJlar5JzQRkUjMBmdRMQN5Q8wSQGzVPAJnUTEAmNZtOqhadVC06qVr0yTIgPwnIpGYCcgNkUvNNQG7UPAFkUvN/OqladFK16KRqEf6RqiUnVYtOqhadVC06qVp0UrXopGrRSdWik6pFJ1WLTqoWnVQtOqladFK16KRq0UnVov8ACbpJRE/izd0AAAAASUVORK5CYII=",
        },
        "description": "Apollos Church",
        "foregroundColor": "rgb(248,247,244)",
        "id": "Pass:ee0dc76d80d041692cdd83b50820a5eb",
        "labelColor": null,
        "logo": Object {
          "uri": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADoAAAA6CAYAAADhu0ooAAAAAXNSR0IArs4c6QAABGFJREFUaAXtWktPU0EUPuX9aHlaeYmIvCkUqEBcGA0GiVFCjAYE6sKdK9f8A3+BiTt3riioiIpBNBrDyoCUt4AaQ4yJUQmPUsrLGUxrezsz9869cyGVns29M2fOzPnmnDn3zGkNcOv2LhwCijgEGPcghoH+b5YOWzRs0RDdgbDrhqjhqGofGotGUbdAIONadSU8uNlJnXFodg4u37tP5Ytg7ItFb9TZmLo2FBdCVpKJOUYrU3egZqMRLpQVM/WMMBigo7aaOUYrU3egbTYrREXIL9NZW6MVC1NeXgOmuDzTLuO23hkqsrPAmpPpbQp/6gq05KgZbLk5ipWWO8uKJyIM1BUor+JttmqIROdVD9INKFa3nTPAZKLI21hSpAdO0A3oucICyE1N4Va6s06foKQb0M56usJ3BoZgdWODuAktVgsYY2OIPC2dugCNj46Gq1UVRL22dnbg7tth6B+fIvJZskQBhZ26AG2xliOrxBJVeDU7D79cLugeHSfycae9np1JUQUZDF2Asr6djtGxPXUGZz7CsttNVO1swUl0vpOJPLWdwoFmmOiRc3N7G/rGp/d09eB3J9l9DegT035KbEooHChWEOeuJHo5MwdL6+s+luOD0/cufbELTgmFA7UzPg/do4HAhjBw1z/g/mBLMzO4sip/WdK7UKCWrEyUr2aR1gHP1hb0T/x1W++ATRSBH49PeptBT97MKmgCvw6hQFmKvZgmBx/HSKCV/XSD1hplNx9/Gdq7MKD4XF63VdHWAdp5fD23AD/X1ohyZpMRmmTuskRBQqcwoOdRlSA7JYmwBIB7cxOeStzWOxAnEI+cdPcVFZSEAWVdnAemZlHK5/FiC3r2jNCTh+bKckiOiwuS4e0QAjQxJgauVFmoa0ujrXTgm/kF+LGyKu3ea8dGRQEurmklIUAxyAQElkQujweeT86QWL6+7d1deOic8LWlLyJuNEKAstz2GQLpQmdUjhyM3PdMQT6cSEuVm4LJ1ww0O9kEOBDRyCFJEmjj3i18hu/LKzS25iqhZqA45cO5KYnWUADCgUgJ7SD37R2jByXWRUHJ/JqBspKE/okpcKOMSCn1MNy30HwE6vNylU4VNE4T0CqU7pWjtI9GctFWKjf86Qt8W1qWdvvaWqyqCSjLmivorjmIknYewn+P6R1jp4TRCorhpDVVA8VlSVyepFGfcxo2ONzWOw+r8pCWmAAXLaXeoVxP1b+mNZYWQUaSkbrY4tKSqtIlDmvYG0yUbAh70RNKvYmqDGKoBmqvtbHmha6mBuiCBuYYNcxLyKKpCfHwm3KPpc2pynVNqPCFC2AHQdGRkdBabeVeWhVQXMqMQyXNgyI1KaEqoKzi9H6AP52fBwXp6VxLcQPFZUj8c8NBE69VuYF2CK7Oqd0wXqDcUbfZUkYtPKtVWo1cOvqm1h4/Bu+/LioSN4T/r6ton0JnEPcZDR1ogZqGgQbuR+i3whYNfRsGIghbNHA/Qr91aCz6BzYbARjcFUOvAAAAAElFTkSuQmCC",
        },
        "logoText": "Apollos Church",
        "passkitFileUrl": null,
        "primaryFields": Array [
          Object {
            "key": "member",
            "label": null,
            "textAlignment": "LEFT",
            "value": "{{ person.FullName }}",
          },
        ],
        "secondaryFields": Array [
          Object {
            "key": "home",
            "label": "home",
            "textAlignment": "LEFT",
            "value": "{{ person | Address:'Home','[[City]], [[State]]' }}",
          },
          Object {
            "key": "campus",
            "label": "campus",
            "textAlignment": "RIGHT",
            "value": "{% assign personCampus = person | Campus %}{{ personCampus.Name  }}",
          },
        ],
        "thumbnail": Object {
          "uri": "https://apollosrock.newspring.cc/GetImage.ashx?guid=dd2aa592-80ac-4d7a-8bdc-47aeb25ce158&format=png&height=200",
        },
        "type": "GENERIC",
      },
    ],
  },
}
`;
